tests/phpunit/tests/blocks/wpBlockPatternsRegistry.php

```diff
@@ -482,4 +482,68 @@ public function test_is_registered_for_known_pattern() {
 		$result = $this->registry->is_registered( 'test/one' );
 		$this->assertTrue( $result );
 	}
+
+	/**
+	 * Ensures theme patterns are registered on init.
+	 *
+	 * @ticket 59723
+	 *
+	 * @covers ::_register_theme_block_patterns
+	 */
+	public function test_register_theme_block_patterns_on_init() {
+		// This test needs to use access static class properties.
+		$registry = WP_Block_Patterns_Registry::get_instance();
+
+		// Ensure we're using a theme with patterns.
+		switch_theme( 'twentytwentythree' );
+
+		$theme          = wp_get_theme();
+		$theme_patterns = array_values( wp_list_pluck( $theme->get_block_patterns(), 'slug' ) );
+
+		// This helper is fired on the init hook.
+		_register_theme_block_patterns();
+
+		$registered = wp_list_pluck( $registry->get_all_registered(), 'name' );
+
+		// Cleanup patterns registry.
+		foreach ( $theme_patterns as $pattern ) {
+			$registry->unregister( $pattern );
+		}
+
+		$this->assertEqualSets( $theme_patterns, array_intersect( $theme_patterns, $registered ), 'Could not confirm theme patterns were registered.' );
```

Comment:
It probably doesn't matter, but would `assertSameSets()` be better here?  That is, use strict typing.

Reply:
I opted for `assertEqualSets()` because I wanted to make sure that the order in which the patterns show up in the registered patterns group wouldn't affect the outcome of the assertion, since we don't care about the order for the purposes of this test.

Reply:
From my reading of the 2 assertions ([assertSameSets()](https://core.trac.wordpress.org/browser/trunk/tests/phpunit/includes/abstract-testcase.php#L983) and [assertEqualSets()](https://core.trac.wordpress.org/browser/trunk/tests/phpunit/includes/abstract-testcase.php#L1003)), order doesn't matter for either.  The only difference is whether the elements of the arrays are compared strictly or loosely.

Reply:
TIL – thanks for linking to the source. I assumed from the name that it operated just like `assertSame()`, but seems like you're correct that order doesn't matter as the lists are both sorted prior to comparing. Updated in [7c42ab5](https://github.com/WordPress/wordpress-develop/pull/5568/commits/7c42ab57f0481f6d46cde6e359697926e1e265f8)